Documentation/ServicingGranularity.md

```diff
@@ -0,0 +1,146 @@
+# Servicing Granularity
+
+This doc is a more detailed description of the problems in [source-build#923
+"Figure out how source-build will work with CoreFX per-package servicing
+policy"](https://github.com/dotnet/source-build/issues/923).
+
+## "Build only if changed"
+
+This is the general rule that if a product hasn't changed, it shouldn't be
+rebuilt and rereleased with a new patch version. It's intuitive and ordinary.
+For example, MSBuild doesn't release new NuGet packages based on downstream
+decisions like a new version of the .NET Core SDK being released.
+
+> A *product* for the sake of this doc is any released asset where an
+independent decision is made whether that specific asset will be patched and
+shipped.
```

Comment:
I think this statement points to the root of our troubles: source code is included in a branch but it is not compiled/released with everything else in the branch.

If the "product" was a separate source repository, I think things would be very much more obvious, as is the case in your nuget/msbuild example above.

Reply:
@nguerrera mentioned in https://github.com/dotnet/core-sdk/issues/5738#issuecomment-558713055 there's some already talk of actually separating out the targeting pack products at least... so there's more hope to get repo-owner-maintained support for "product subsets" than I thought. I'll update the doc to reflect this.

We can also take different approaches for different products where practical. I mention the CoreFX OOB packages a lot here, and they mess up some approaches, but targeting packs have constraints (only one used per TFM per release) that make them better candidates for this.

Reply:
Another approach for targeting packs in particular is generating them in [source-build-reference-packages](https://github.com/dotnet/source-build-reference-packages), where the source code will be static for sure. This creates some maintenance load because when there's a new TFM or patch, the new version will have to be reverse engineered. But it's a known process that the source-build team is already using for the Microsoft.NETCore.App.Ref 3.0.0 package in the 3.0.1 build. (I learned about this today.)

I think it makes sense to go with that for targeting packs until we get a feature in the repos to let us easily go back in time and specifically build the targeting packs.

I'm going to try narrowing this doc down to focus on the CoreFX OOB packages.